Components/LearningCore/Source/gov/sandia/cognition/learning/algorithm/clustering/SNNClusterer.java

```diff
@@ -0,0 +1,637 @@
+    {
+        this.k = k;
+    }
+
+    /**
+     * Gets the number of nearest neighbors.
+     *
+     * @return k the number of nearest neighbors.
+     */
+    public int getNumNeighbors()
+    {
+        return this.k;
+    }
+
+    /**
+     * Gets the number of clusters.
+     *
+     * @return the cluster count.
+     */
+    public int getClusterCount()
+    {
+        return this.clusterCount;
+    }
+
+    /**
+     * Sets the similarity score (mutually shared nearest neighbors) threshold.
+     *
+     * @param eps The minimum similarity score required.
+     */
+    public void setSimilarity(int eps)
+    {
+        this.eps = eps;
+    }
+
+    /**
+     * Gets the similarity score threshold (eps).
+     *
+     * @return eps
+     */
+    public int getSimiliarity()
+    {
+        return this.eps;
+    }
+
+    /**
+     * Sets the minimum neighboring points to have a similarity score of at
+     * least eps for the given point to be considered a core point.
+     *
+     * @param minPts The minimum points threshold.
+     */
+    public void setMinPts(int minPts)
+    {
+        this.minPts = minPts;
+    }
+
+    /**
+     * Gets the minPts threshold to be considered a core point.
+     *
+     * @return minPts
+     */
+    public int getMinPts()
+    {
+        return this.minPts;
+    }
+
+    /**
+     * Sets the distance metric the clustering uses.
+     *
+     * @param metric The metric.
+     */
+    public void setMetric(Semimetric<? super DataType> metric)
+    {
+        this.metric = metric;
+    }
+
+    /**
+     * Gets the distance metric the clustering uses.
+     *
+     * @return The metric.
+     */
+    public Semimetric<? super DataType> getMetric()
+    {
+        return this.metric;
+    }
+
+    /**
+     * Gets the current clusters, which is a sparse mapping of exemplar
+     * identifier to cluster object.
+     *
+     * @return The current clusters.
+     */
+    protected ArrayList<ClusterType> getClusters()
+    {
+        return this.clusters;
+    }
+
+    /**
+     * Get the cluster at this index.
+     *
+     * @param i The index of the cluster.
+     * @return
+     */
+    public ClusterType getCluster(int i)
+    {
+        return this.getClusters().get(i);
+    }
+
+    /**
+     * Sets the cluster creator.
+     *
+     * @param creator The creator for clusters.
+     */
+    public void setCreator(
+        ClusterCreator<ClusterType, DataType> creator)
+    {
+        this.creator = creator;
+    }
+
+    /**
+     * Gets the cluster creator.
+     *
+     * @return The cluster creator.
+     */
+    public ClusterCreator<ClusterType, DataType> getCreator()
+    {
+        return this.creator;
+    }
+
+    /**
+     * SemimetricPointComparator acts as a comparator for pairs of points where
+     * each point also has the distance (a double given by the semimetric
+     * evaluate method) between that point and a reference point.
+     */
+    private class SemimetricPointComparator

```

Comment:
You can probably replace this class with just using a lambda expression above and use Double.compare to make it a one-liner.